lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
+          unless what.is_a? String
+            if what.class.find_by_id(what.id)
```

Comment:
Yes, I didn't want to give a message with objects from a different scope.
Maybe a message giving the object name and an out of scope explanation is better - "example.com is being used by Host::Managed that you can't see"?